ipa-core/src/protocol/ipa_prf/malicious_security/quadratic_proofs.rs

```diff
@@ -0,0 +1,933 @@
+use futures::StreamExt;
+use futures_util::{stream, TryStreamExt};
+use ipa_macros::Step;
+
+use crate::{
+    error::Error,
+    ff::{Field, Invert},
+    helpers::{Direction, ReceivingEnd, SendingEnd},
+    protocol::{
+        basics::validate_replicated_shares::validate_sum_to_zero,
+        context::Context,
+        ipa_prf::malicious_security::{
+            fiat_shamir::{fiat_shamir_prover, fiat_shamir_verifier},
+            lagrange::{
+                compute_lagrange_base, compute_lagrange_denominator, generate_evaluation_points,
+                lagrange_evaluation, lagrange_evaluation_precomputed,
+            },
+        },
+        prss::SharedRandomness,
+        RecordId,
+    },
+    seq_join::seq_join,
+};
+
+#[derive(Step)]
+pub(crate) enum Step {
+    HashFromLeft,
+    HashFromRight,
+    PQFromLeft,
+    TwoOutOfTwoZeroCheck,
+}
+
+/// Non-Interactive Distributed Zero Knowledge Proof
```

Comment:
Please add a reference to the paper. I'd prefer to use `https://eprint.iacr.org/2023/909.pdf` and point to section 4.1. 

It may also be worth pointing (additionally) to the paper that introduced this technique: section 3 of `https://eprint.iacr.org/2020/1451.pdf`